### PR TITLE
JavaDoc and various tweaks for ValueHintProvider

### PR DESCRIPTION
**Pull request by pperalta**

### CODE REVIEW

File: spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/ValueHintProvider.java [hunk 1]
```diff
@@ -22,19 +22,34 @@
...
  * @author Eric Bottard
  */
 public interface ValueHintProvider {
 
-	List<ValueHint> guessValueHints(ConfigurationMetadataProperty property, ClassLoader classLoader);
+	/**
+	 * For a given property, return a list of {@link ValueHint} that may apply.
+	 *
+	 * @param property     property for which to generate value hints
+	 * @param classLoader  class loader for the artifact/module that this
```
Comment (pperalta) [inline spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/ValueHintProvider.java]: Tried to explain what this `ClassLoader` is for; hopefully this makes sense and the terminology is correct in this context.

File: spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/BooleanValueHintProvider.java [hunk 2]
```diff
@@ -42,12 +42,10 @@
...
 	@Override
-	public List<ValueHint> guessValueHints(ConfigurationMetadataProperty property, ClassLoader classLoader) {
-		if ("java.lang.Boolean".equals(property.getType())) {
-			return BOOLEANS;
-		} else {
-			return Collections.emptyList();
-		}
+	public List<ValueHint> generateValueHints(ConfigurationMetadataProperty property, ClassLoader classLoader) {
+		return "java.lang.Boolean".equals(property.getType())
+				? Collections.unmodifiableList(BOOLEANS)
```
Comment (ericbottard) [inline spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/BooleanValueHintProvider.java]: Can move the unmodifiable wrapping in the initializer. Will do on merge

**Comment by ericbottard** [comment]
Merged with minor tweak as 143d5c5

File: spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/BooleanValueHintProvider.java (hunk 2 above)
Comment (pperalta) [inline spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/BooleanValueHintProvider.java]: +1

**Pull request closed without merging**